checker/src/tag_domain.rs

```diff
@@ -97,20 +97,35 @@ pub struct Tag {
 
 /// An element of the tag domain is essentially an over-approximation for present and absent tags.
 /// The approximation is denoted as a map from tags to lifted Boolean values (`BoolDomain`).
+/// Unrecorded tags are mapped to the `default` field of the tag domain element.
 /// If a tag is mapped to `True`, then it must be present.
 /// If a tag is mapped to `False', then it must be absent.
 /// If a tag is mapped to `Top`, then it may or may not be present.
 #[derive(Ord, PartialOrd, Eq, PartialEq, Debug, Clone, Serialize, Deserialize)]
 pub struct TagDomain {
     map: RedBlackTreeMap<Tag, BoolDomain>,
+    default: BoolDomain,
```

Comment:
I'm none too sure about this. Why not just add a (tag, top) entry to map?

Reply:
The default field is used to answer tag queries. If a tag is kept track in the map, we just look it up in the map; otherwise, we use the default as the check result.